framework/Objects/Post_Type.php

```diff
@@ -333,4 +337,13 @@ public function views( $template ) {
 
 		return $template;
 	}
+
+	/**
+	 * Init FakerPress.
+	 */
+	public function init_faker() {
+		if ( method_exists( $this, 'faker' ) ) {
+			new FakerPress( $this::$ID, $this );
```

Comment:
if you use singleton for FakerPress - you can't run "new " it should generate a fatal.
singleton pattern have to have protected constructor method